waku/waku_lightpush/protocol.nim

```diff
@@ -26,6 +26,25 @@ type WakuLightPush* = ref object of LPProtocol
   pushHandler*: PushMessageHandler
   requestRateLimiter*: Option[TokenBucket]
 
+proc validateMessage*(
```

Comment:
Ah, the idea is to use the exact same validation function that Relay would use when it publishes the message - not create a new validation function (which would have to be maintained separately and would invariably at some point miss validation edge cases enforced by the relay validator).
In other words, we should use: https://github.com/waku-org/nwaku/blob/b46226fca0fbc5730cbf3c9899e33fac8b817796/waku/waku_relay/protocol.nim#L225-L252

Of course, that requires a reference to the instantiated `wakuRelay` object that will eventually publish this message. This is currently not available in the lightpush module, but in the pushHandler provided to lightpush by the waku node here: https://github.com/waku-org/nwaku/blob/b46226fca0fbc5730cbf3c9899e33fac8b817796/waku/node/waku_node.nim#L917-L929

Architecturally, I think this indirection created by the pushHandler is unnecessary. I'll add a comment on this for future work, but for now I think this PR should be as simple as adding:
```
node.wakuRelay.validateMessage(
```
in the push handler before pushing the message (and testing)

Reply:
@shash256 and @jm-clius : I think the biggest issue with lightpush is not that it does not pre-validates the message but in contrary it will not get noticed by the relay that the message validation done in relay protocol side is failed. It cause that we do not have meaningful answer to the lightpush client that would help error handling.
WDYT?

Reply:
Well, if we validate the message before attempting to relay it and return an explicit "validation error" to the client on failure, wouldn't it cover - or at least improve - this silent failure? Similar to how we return a "bad request" error to Relay API clients in REST: https://github.com/waku-org/nwaku/blob/master/waku/waku_api/rest/relay/handlers.nim#L163-L164